### PR TITLE
PMM-2643: Use gopsutils with unix socket support.

### PR DESCRIPTION
**Pull request by arvenil**

### CODE REVIEW

File: dsn/dsn.go [hunk 1]
```diff
@@ -210,25 +211,40 @@ func GetSocketFromProcessLists() (string, error) {
...
 		return "", errors.Wrap(err, "Cannot get the list of PIDs")
 	}
+	sockets := []string{}
 	for _, pid := range pids {
 		proc, err := process.NewProcess(pid)
 		if err != nil {
 			continue
 		}
-		if procName, err := proc.Name(); err != nil {
+		procName, err := proc.Name();
```
Comment (golangci) [inline dsn/dsn.go]: File is not goimports-ed

**Comment by arvenil** [comment]
aaaaand it doesn't work

File: dsn/lsof/lsof.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	p := Process{}
+	for index, line := range lines {
+		if strings.HasPrefix(line, "f") {
+			err := p.parseFileLines(lines[index:])
+			if err != nil {
+				return p, err
+			}
+			break
+		} else {
+			p.fillField(line)
```
Comment (golangci) [inline dsn/lsof/lsof.go]: Error return value of `p.fillField` is not checked

File: dsn/dsn.go [hunk 3]
```diff
@@ -200,6 +203,29 @@ func HidePassword(dsn string) string {
 	return dsn
 }
 
+// GetSocketFromTCPConnection will try to get socket path by connecting to MySQL localhost TCP port.
+// This is not reliable as TCP connections may be not allowed.
+func GetSocketFromTCPConnection(dsn DSN) (socket string, err error) {
```
Comment (golangci) [inline dsn/dsn.go]: dsn can be fmt.Stringer

File: dsn/dsn.go [hunk 4]
```diff
@@ -210,28 +236,102 @@ func GetSocketFromProcessLists() (string, error) {
...
+		log.Println("netstat: multiple sockets detected, choosing first one:", strings.Join(sockets, ", "))
+	}
+	if len(sockets) > 0 {
+		return sockets[0], nil
+	}
 	return "", ErrNoSocket
 }
 
+// GetSocket tries to detect and return path to the MySQL socket.
+func GetSocket(dsn DSN) (string, error) {
```
Comment (golangci) [inline dsn/dsn.go]: dsn can be fmt.Stringer

**Pull request merged**